src/main/java/net/snowflake/client/core/SFBaseStatement.java

```diff
@@ -168,4 +169,7 @@ public void executeSetProperty(final String sql) {
   public abstract long getConservativeMemoryLimit();
 
   public abstract int getConservativePrefetchThreads();
+
+  /** @return the query IDs of the latest executed batch queries */
+  public abstract List<String> getChildQueryIds(String queryID) throws SQLException;
```

Comment:
Hey @sfc-gh-ema and @sfc-gh-zli We are adding a new API to `SFBaseStatement`. It is fine to throw `UnsupportedOperationException` for this case since it will be used only by the new API `df.collectAsync`